src/firewall/functions.py

```diff
@@ -43,6 +43,43 @@ def fail(msg=None):
 ###############################################################################
 
 
+def addr_family(family, allow_unspec=False):
+    """Normalizes (and verifies) the address family. It returns the corresponding
+    socket.AF_INET, socket.AF_INET6, or socket.AF_UNSPE
+
+    @family: the family to normalize/verify.
+    @allow_unspec: whether socket.AF_UNSPEC and None is accepted as socket.AF_UNSPE
+    """
+    if family == socket.AF_INET or family == socket.AF_INET6:
+        return family
+    if family is None or family == socket.AF_UNSPEC:
+        if allow_unspec:
+            return socket.AF_UNSPEC
```

Comment:
Because a None/ANY/UNSPEC address family makes sense in some cases.

For example, `ipaddr_parse()` you can tell to accept either Ipv4/IPv6, or determine which address family must be allowed.

However, by default it is rejected as invalid. The caller has to explicitly opt-in via `allow_unspec=True`.